source/grid/grid_tools.cc

```diff
@@ -2812,8 +2812,8 @@ namespace GridTools
       if (cell->active())
         {
           while (current_cell_idx >=
-                 std::floor((long)n_active_cells * (current_proc_idx + 1) /
-                            n_partitions))
+                 std::floor(static_cast<long>(n_active_cells) *
```

Comment:
I think we want something that's guaranteed to be 64-bit here. Maybe use `std::uint64_t`?